src/components/ha-wa-dialog.ts

```diff
@@ -0,0 +1,380 @@
+  public headerSubtitle = "";
+
+  @property({ type: String, attribute: "header-subtitle-position" })
+  public headerSubtitlePosition: "above" | "below" = "below";
+
+  @property({ type: Boolean, reflect: true, attribute: "flexcontent" })
+  public flexContent = false;
+
+  @state()
+  private _open = false;
+
+  @state()
+  private _sizeChanged = false;
+
+  protected updated(
+    changedProperties: Map<string | number | symbol, unknown>
+  ): void {
+    super.updated(changedProperties);
+
+    if (changedProperties.has("open")) {
+      this._open = this.open;
+    }
+
+    if (changedProperties.has("width")) {
+      this._sizeChanged = false;
+    }
+
+    this.classList.toggle("size-changed", this._sizeChanged);
+  }
+
+  protected render() {
+    return html`
+      <wa-dialog
+        .open=${this._open}
+        .lightDismiss=${!this.preventScrimClose}
+        without-header
+        @wa-show=${this._handleShow}
+        @wa-after-hide=${this._handleAfterHide}
+      >
+        <slot name="heading">
+          <ha-dialog-header .subtitlePosition=${this.headerSubtitlePosition}>
+            <slot name="navigationIcon" slot="navigationIcon">
+              <ha-icon-button
+                data-dialog="close"
+                .label=${this.hass?.localize("ui.common.close") ?? "Close"}
+                .path=${mdiClose}
+              ></ha-icon-button>
+            </slot>
+            ${this.headerTitle
+              ? html`<span
+                  slot="title"
+                  class="title"
+                  @click=${this.toggleWidth}
+                >
+                  ${this.headerTitle}
+                </span>`
+              : nothing}
+            ${this.headerSubtitle
+              ? html`<span slot="subtitle">${this.headerSubtitle}</span>`
+              : nothing}
+            <slot name="actionItems" slot="actionItems"></slot>
+          </ha-dialog-header>
+        </slot>
+        <div class="body ha-scrollbar">
+          <slot></slot>
+        </div>
+        <slot name="footer" slot="footer"></slot>
+      </wa-dialog>
+    `;
+  }
+
+  private _handleShow = () => {
+    this._open = true;
+    fireEvent(this, "opened");
+
+    this.updateComplete.then(() => {
+      const focusElement = this.querySelector(
+        "[dialogInitialFocus]"
+      ) as HTMLElement;
+      focusElement?.focus();
+    });
+  };
+
+  private _handleAfterHide = () => {
+    this._open = false;
+    fireEvent(this, "closed");
+  };
+
+  public disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this._open = false;
+  }
+
+  public toggleWidth = () => {
+    if (this.widthOnTitleClick === "none") {
+      return;
+    }
+
+    this._sizeChanged = !this._sizeChanged;
+  };
+
+  static styles = [
+    haStyleScrollbar,
+    css`
+      :host([scrolled]) wa-dialog::part(header) {
+        max-width: 100%;
+        border-bottom: 1px solid
+          var(--dialog-scroll-divider-color, var(--divider-color));
+      }
```

Comment:
I agree we should add this. But this is using the wa-dialog header not our slot.